jvm-runtime/plugin/common/service.go

```diff
@@ -119,24 +118,23 @@ func (s *Service) SyncStubReferences(ctx context.Context, req *connect.Request[l
 	return connect.NewResponse(&langpb.SyncStubReferencesResponse{}), nil
 }
 
-// Build the module and stream back build events.
-//
-// A BuildSuccess or BuildFailure event must be streamed back with the request's context id to indicate the
-// end of the build.
-//
-// The request can include the option to "rebuild_automatically". In this case the plugin should watch for
-// file changes and automatically rebuild as needed as long as this build request is alive. Each automactic
-// rebuild must include the latest build context id provided by the request or subsequent BuildContextUpdated
-// calls.
 func (s *Service) Build(ctx context.Context, req *connect.Request[langpb.BuildRequest]) (*connect.Response[langpb.BuildResponse], error) {
 	logger := log.FromContext(ctx)
 	_ = os.Setenv("QUARKUS_ANALYTICS_DISABLED", "true") //nolint:errcheck
 	logger = logger.Module(req.Msg.BuildContext.ModuleConfig.Name)
 	ctx = log.ContextWithLogger(ctx, logger)
+
 	buildCtx, err := buildContextFromProto(req.Msg.BuildContext)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+
+	release, err := flock.Acquire(ctx, buildCtx.Config.BuildLock, BuildLockTimeout)
```

Comment:
build lock is now locked on each build, not just the first build in `runQuarkusDev`

Reply:
Do we even need the build lock now?